### PR TITLE
fix registration, move invites logic, fix 'forgot pasword' page markup

### PR DESCRIPTION
**Pull request by vikdotdev**
- [x] merged with the latest master branch
- [x] resolved all the conflicts, PR could be merged
- [x] new code has new specs
- [x] for new models - new factories were added too
- [x] unit specs are passed
- [x] code is well-formatted, Ruby, CSS, etc
- [x] screenshots of new pages, features are added
- [x] code passed internal team code review
- [x] no instance variables in partials
- [x] code is well formatted and has spaces!!!

PS: delete your branch in github after merge http://prntscr.com/q80mvq

### CODE REVIEW

**Comment by vikdotdev** [comment]
2019-12-13T13:33:59
2019-12-13T13:33:48

File: app/controllers/registrations_controller.rb [hunk 1]
```diff
@@ -0,0 +1,62 @@
+class RegistrationsController < ApplicationController
```
Comment (Sweex99) [inline app/controllers/registrations_controller.rb]: InvitationsController
2 actions
open + accept

File: app/models/user.rb [hunk 2]
```diff
@@ -52,4 +51,8 @@ def self.to_csv
       end
     end
   end
+
+  def valid_disregarding_organization?
```
Comment (Sweex99) [inline app/models/user.rb]: remove

File: app/controllers/registrations_controller.rb [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  layout 'devise'
+
+  def new
+    @token = params[:invite_token]
+    @user = User.new
+    @user.organization = Organization.new
```
Comment (Sweex99) [inline app/controllers/registrations_controller.rb]: remove @user.organization = Organization.new

File: app/controllers/registrations_controller.rb [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    end
+  end
+
+  def accept
+    @token = params[:user][:invite_token]
+
+    return if @token.nil?
+
+    @user = User.new(sign_up_params)
+    invite = Invite.find_by_token(@token)
```
Comment (Sweex99) [inline app/controllers/registrations_controller.rb]: find_by(token: @token)

Comment (Sweex99) [reply app/controllers/registrations_controller.rb]: find_by!(...)

File: app/controllers/registrations_controller.rb [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    @user.organization = invite.organization
+    @user.email = invite.email
+    @user.role = invite.role
+
+    if @user.save
+      invite.destroy
+      bypass_sign_in @user
+      flash[:success] = 'User successfully created.'
+      redirect_to account_user_path(@user)
```
Comment (Sweex99) [inline app/controllers/registrations_controller.rb]: dashboard_path

File: app/controllers/accept_invites_controller.rb [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+  def new
+    @token = params[:invite_token]
+    @user = User.new
+  end
+
+  def create
+    @token = params[:user][:invite_token]
+
+    return if @token.nil?
```
Comment (igorkasyanchuk) [inline app/controllers/accept_invites_controller.rb]: що значить return 
видаляй це, воно користі немає

File: app/controllers/accept_invites_controller.rb [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+    @user = User.new(accept_params)
+    invite = Invite.find_by(token: @token)
```
Comment (igorkasyanchuk) [inline app/controllers/accept_invites_controller.rb]: "find_by!" має бути і я спитаю чого
я вже тоді це казав

Comment (vikdotdev) [reply app/controllers/accept_invites_controller.rb]: поняв, прочитав, поправив

File: app/models/user.rb [hunk 8]
```diff
@@ -52,4 +54,14 @@ def self.to_csv
...
   end
+
+  private
+
+  def organization_validation
+    self.skip_organization_validation ||= false
+    return unless organization_id.nil?
+    return if super_admin? || skip_organization_validation
+
+    errors.add(:organization_id, 'should be present')
```
Comment (igorkasyanchuk) [inline app/models/user.rb]: must be present

**Pull request merged**